lib/src/animated_map_controller.dart

```diff
@@ -123,83 +133,156 @@ class AnimatedMapController extends MapControllerImpl {
     AnimationId animationId = AnimationId(
       destLocation: effectiveDest,
       destZoom: effectiveZoom,
+      customId: customId,
     );
 
     bool hasTriggeredMove = false;
+    bool animationCompleted = false;
 
     animationController.addListener(() {
+      // The animation calls this listener with value 1.0 twice. Once when the
+      // value is 1.0 but isCompleted is false and again when it is still 1.0
+      // and isCompleted is true. This check ensures we don't trigger a
+      // duplicate movement but also, more importantly, that we trigger the
+      // final movement with the finished id exactly once.
+      if (animationCompleted) return;
+      animationCompleted |= animation.value == 1.0;
+
       animationId = animationId.copyWith(
         moveId: AnimatedMoveId.fromAnimationAndTriggeredMove(
-          animationValue: animation.value,
+          animationIsCompleted: animationCompleted,
           hasTriggeredMove: hasTriggeredMove,
         ),
       );
 
-      hasTriggeredMove |= move(
-        latLngTween.evaluate(animation),
-        zoomTween.evaluate(animation),
-        id: animationId.id,
+      hasTriggeredMove |= movementCallback(
+        animation,
+        latLngTween,
+        zoomTween,
+        rotateTween,
+        animationId,
       );
-      rotate(rotateTween.evaluate(animation));
     });
 
     return animationController.forward();
   }
 
+  // Determine what MapController method should be called based on whether
+  // there is movement and/or rotation. If there is neither movement nor
+  // rotation null is returned.
+  bool Function(
+    CurvedAnimation animation,
+    LatLngTween latLngTween,
+    Tween<double> zoomTween,
+    Tween<double> rotateTween,
+    AnimationId animationId,
+  )? _movementCallback({
```

Comment:
I think it would be more readable to use a typedef:
```dart
typedef _MovementCallback = bool Function(
  CurvedAnimation animation,
  LatLngTween latLngTween,
  Tween<double> zoomTween,
  Tween<double> rotateTween,
  AnimationId animationId,
);
```

Reply:
Definitely 👍